src/flag_gems/utils/libentry.py

```diff
@@ -27,6 +28,17 @@
 version = triton.__version__.split(".")
 major_version, minor_version = eval(version[0]), eval(version[1])
 
+
+def get_kernel_hash(func):
+    if hasattr(func, "fn"):
+        original_func = func.fn
+    else:
+        original_func = func
+
+    source_code = inspect.getsource(original_func)
+    return hashlib.md5(source_code.encode("utf-8")).hexdigest()[:8]
```

Comment:
I suggest also storing the hash key of the configuration space. If the configuration space expands or shrinks, the optimal configuration may change.